documentation/docs/extensibility.md

```diff
@@ -1,86 +1,199 @@
-      params.originalStage()
+The centrally maintained pipeline checks if such a file exists and executes it, if present.
+A parameter of type `Map` that contains the following keys is passed to the extension:
 
-      //access overall pipeline script object
-      echo "Branch: ${params.script.commonPipelineEnvironment.gitBranch}"
+* `script`: defines the global script environment of the `Jenkinsfile` run. This makes sure that the correct configuration environment can be passed to project "Piper" steps and allows access to for example the `commonPipelineEnvironment`.
+* `originalStage`: this will allow you to execute the "original" stage at any place in your script. If omitting a call to `originalStage()` only your code will be executed instead.
+* `stageName`: name of the current stage
+* `config`: configuration of the stage and general config (including all defaults)
 
-      echo "End - Extension for stage: ${params.stageName}"
-  }
-  return this
-  ```
+Here is a simple example for such an extension, which you can use as a starting point:
+
+```groovy
+void call(Map params) {
+  //access stage name
+  echo "Start - Extension for stage: ${params.stageName}"
+
+  //access config
+  echo "Current stage config: ${params.config}"
+
+  //execute original stage as defined in the template
+  params.originalStage()
+
+  //access overall pipeline script object
+  echo "Branch: ${params.script.commonPipelineEnvironment.gitBranch}"
+
+  echo "End - Extension for stage: ${params.stageName}"
+}
+return this
+```
+
+!!! note "`return this`"
+    Don't forget the `return this` which is required at the end of _all_ extension scripts.
+    This is due to how Groovy loads scripts internally.
 
 !!! note "Init stage cannot be extended"
-    Please note, the `Init` stage among other things also checks out your current repository.<br />Thus it is not possible to use extensions on this stage.
+    Please note, the `Init` stage among other things also checks out your current repository including your extensions.<br />
+    Thus it is not possible to use extensions on this stage.
+
+### Practical example
+
+For a more practical example, you can use extensions in SAP Cloud SDK Pipeline to add custom linters to the pipeline.
+
+A linter is a tool that can check the source code for certain stylistic criteria, and many teams chose to use a linter to ensure a common programming style.
+
+As an example, if you want to use [Checkstyle](https://checkstyle.sourceforge.io/) in your codebase, you might use an extension similar to this one in a file called `.pipeline/extensions/lint.groovy` in your project:
+
+```groovy
+def call(Map parameters) {
+
+    parameters.originalStage() // Runs the built in linters
+
+    mavenExecute(
+        script: parameters.script,
+        flags: '--batch-mode',
+        pomPath: 'application/pom.xml',
+        m2Path: s4SdkGlobals.m2Directory,
+        goals: 'checkstyle:checkstyle',
+    )
+
+    recordIssues blameDisabled: true,
```

Comment:
Add comment explaining what `recordIssues` would do here.

Reply:
I added some text after this code snippet explaining what is going on. I think this is better compared to a code comment. Thanks for the suggestion.